identity-wallet/src/state/credentials/reducers/send_credential_request.rs

```diff
@@ -136,39 +127,61 @@ pub async fn send_credential_request(mut state: AppState, action: Action) -> Res
+                    .get(&credential_configuration_id)
+                    .ok_or(UnknownCredentialConfigurationIdError(
+                        credential_configuration_id.clone(),
+                    ))?
+                    .credential_format
+                    .to_owned();
+
                 // Get the credential.
                 let credential_response = wallet
-                    .get_credential(
-                        credential_issuer_metadata,
-                        &token_response,
-                        credential_offer_formats[0].to_owned(),
-                    )
+                    .get_credential(credential_issuer_metadata, &token_response, credential_format)
                     .await
                     .map_err(GetCredentialError)?;
 
                 let credential = match credential_response.credential {
-                    CredentialResponseType::Immediate(credential) => credential,
+                    CredentialResponseType::Immediate { credential, .. } => credential,
                     _ => panic!("Credential was not a jwt_vc_json."),
                 };
 
-                vec![credential]
+                vec![(credential_configuration_id, credential)]
             }
             _batch => {
+                let (credential_configuration_ids, credential_formats): (Vec<_>, Vec<_>) =
+                    credential_configurations_supported
+                        .into_iter()
+                        .map(|(credential_configuration_id, credential_configuration)| {
+                            (credential_configuration_id, credential_configuration.credential_format)
+                        })
+                        .unzip();
+
                 let batch_credential_response = wallet
-                    .get_batch_credential(credential_issuer_metadata, &token_response, credential_offer_formats)
+                    .get_batch_credential(credential_issuer_metadata, &token_response, credential_formats)
                     .await
                     .map_err(GetBatchCredentialError)?;
 
-                batch_credential_response
-                    .credential_responses
+                credential_configuration_ids
                     .into_iter()
-                    .map(|credential_response| match credential_response {
-                        CredentialResponseType::Immediate(credential) => credential,
-                        _ => panic!("Credential was not a jwt_vc_json."),
-                    })
+                    .zip(batch_credential_response.credential_responses.into_iter())
+                    .filter_map(
+                        |(credential_configuration_id, credential_response)| match credential_response {
+                            CredentialResponseType::Immediate { credential, .. } => {
+                                Some((credential_configuration_id, credential))
+                            }
+                            // TODO: add support for deffered credentials.
```

Comment:
Is this a todo for next PR, or still needs implementation?

Reply:
This will be implemented in a future PR :)